### ISSUE TITLE
`rails destroy graphql:mutation` removes mutations folder and mutation_type.rb

### ISSUE THREAD

**Issue by cpunion** [rmosolgo/graphql-ruby#1099]
Steps:

```console
MBP:backend lijie$ rails g graphql:mutation Foo
Running via Spring preloader in process 91374
       exist  app/graphql/mutations
      create  app/graphql/mutations/.keep
      create  app/graphql/types/mutation_type.rb
add_root_type  mutation
      create  app/graphql/mutations/foo.rb
        gsub  app/graphql/types/mutation_type.rb

MBP:backend lijie$ rails g graphql:mutation Bar
Running via Spring preloader in process 91378
       exist  app/graphql/mutations
   identical  app/graphql/mutations/.keep
        skip  app/graphql/types/mutation_type.rb
add_root_type  mutation
      create  app/graphql/mutations/bar.rb
        gsub  app/graphql/types/mutation_type.rb

MBP:backend lijie$ rails d graphql:mutation Bar
Running via Spring preloader in process 91567
      remove  app/graphql/mutations
      remove  app/graphql/mutations/.keep
      remove  app/graphql/types/mutation_type.rb
add_root_type  mutation
      remove  app/graphql/mutations/bar.rb
```

Expects: Don't remove mutations folder and mutation_type.rb (At least keep it if I have some mutation types)

**Comment by Willianvdv** [comment]
@cpunion interesting! I never heard about the `rails destroy` (http://guides.rubyonrails.org/command_line.html#rails-destroy) command before.

Do you think you can send in a PR to get this fixed?

**Comment by cpunion** [comment]
@Willianvdv Yay, I will try.

**Comment by thefliik** [comment]
Uggg. I had to update the generator of another gem I use to properly accommodate `rails destroy`. I found the documentation surrounding `rails destroy` / generators in general to be rather poor and the whole process took me much longer than it should have. Let me know, @cpunion, if you run into issues figuring it out. I _may_ be able to help.

**Comment by cpunion** [comment]
@thefliik Awesome. I feel I can't figure out how to make the generators full reversible quickly. So I'm very happy if you can fix it or help. 😃

**Comment by thefliik** [comment]
I'll try to look into it this weekend. Anyone know if there's a way for me to mark this issue as "unread" / make it show up in Github's notifications panel so that I don't forget about it?

**Comment by Willianvdv** [comment]
@thefliik no, but I can ping you ;)

**Comment by thefliik** [comment]
I think I fixed this issue in PR #1119. I also left some comments about how rails' generators work / how `rails destroy` is built.

**Comment by rmosolgo** [comment]
Fixed by #1119
